include/ppx/application.h

```diff
@@ -209,8 +209,25 @@ struct ApplicationSettings
     bool enableDisplay         = true;
     bool enableImGui           = false;
     bool allowThirdPartyAssets = false;
-    bool enableXR              = false;
-    bool enableXRDebugCapture  = false;
+
+    struct
+    {
+        bool enable             = false;
+        bool enableDebugCapture = false;
+
+        float depthNearPlane = 0.001f;
```

Comment:
this is not really xr specific values (and they change at per-application bases), better to be consistent with non-xr code when we create projection matrix. Either we apply this setting for all regular projection matrices, or maybe we can just use the value from regular projecting matrices at per application bases.

Reply:
The problem with not having them as part of the XR settings is that then there needs to be a way to pass those values to the XR begin- or end-of-frame routine. The depth info layer needs these values at every frame. I don't currently see a way to modify the render loop to have an application pass these values cleanly. 

A quick way would be to add an application-level method that the application must call every frame like `SetXrNearFarPlanes()`, but I'm not a fan. 

What do you think?

Reply:
me neither, adding the extra call requires the application to know where to call this function, and calling it after BeginFrame would break the code.

How about keep the GetProjectionMatrixForCurrentView with near/far plane as input (as it is now), but rename it as GetProjectionMatrixForCurrentViewAndSetXrNearFarPlanes. And in XrComponent, we keep near/far as members, and move updating mCompositionLayerProjectionViews & & mCompositionLayerQuad & mCompositionLayerDepthInfos to EndFrame?

Agree it is still a bit ugly, and I am also open to other options. :) the reason why i mentioned it here is because that it is normal to see application have different near/far planes when they create projection matrix, with this setup, we force the application to use [0.001f, 10000.0f], if they don't, there might be some bugs that could be hard to debug.

Reply:
> How about keep the GetProjectionMatrixForCurrentView with near/far plane as input (as it is now), but rename it as GetProjectionMatrixForCurrentViewAndSetXrNearFarPlanes. And in XrComponent, we keep near/far as members, and move updating mCompositionLayerProjectionViews & & mCompositionLayerQuad & mCompositionLayerDepthInfos to EndFrame?

I think I will go for this one. It still is a bit dodgy if somebody calls `GetProjectionMatrixForCurrentViewAndSetXrNearFarPlanes` multiple times with different values, but I will add an assert to make sure this doesn't happen.

We can modify this once we have a better solution.

Thanks!

Reply:
sounds good. thanks :)